crates/services/gas_price_service/src/fuel_gas_price_updater/fuel_da_source_adapter.rs

```diff
@@ -1,16 +1,125 @@
 use crate::fuel_gas_price_updater::{
-    DaCommitDetails,
-    DaCommitSource,
+    DaBlockCosts,
+    Error::CouldNotFetchDARecord,
+    GetDaBlockCosts,
     Result as GasPriceUpdaterResult,
 };
+use anyhow::anyhow;
+use std::sync::Arc;
+use tokio::sync::Mutex;
 
-#[derive(Default, Clone)]
-pub struct FuelDaSource;
+pub mod block_committer_costs;
+pub mod dummy_costs;
+pub mod service;
 
-impl DaCommitSource for FuelDaSource {
-    fn get_da_commit_details(
-        &mut self,
-    ) -> GasPriceUpdaterResult<Option<DaCommitDetails>> {
-        todo!() // TODO(#2139): pending research on how to get the data from the block committer
+pub use block_committer_costs::BlockCommitterDaBlockCosts;
+pub use dummy_costs::DummyDaBlockCosts;
+pub use service::*;
+
+pub const POLLING_INTERVAL_MS: u64 = 10_000;
+
+pub type DaBlockCostsProvider = Arc<Mutex<Option<DaBlockCosts>>>;
+
+impl GetDaBlockCosts for DaBlockCostsProvider {
+    fn get(&mut self) -> GasPriceUpdaterResult<Option<DaBlockCosts>> {
+        let mut da_block_costs_guard = self.try_lock().map_err(|err| {
```

Comment:
we don't want it to be an expensive operation in case the lock is held by the function that sets data into it, and we always want to be able to return instantly. as per discussion with @MitchTurner